code/game/objects/structures/candybucket.dm

```diff
@@ -0,0 +1,82 @@
+/obj/structure/candybucket
+	name = "candy bucket"
+	icon = 'icons/obj/candybucket.dmi'
+	icon_state = "jackbucket_empty"
+	desc = "A thematically appropriate bucket for filling with candy or other goodies."
+	density = 1
+	anchored = 1
+	var/bucketType = "jackbucket"
+	var/candypacity = 30
+	var/spooky = FALSE
+	var/list/allowedTreats = list(
+		/obj/item/weapon/reagent_containers/pill,
+		/obj/item/weapon/reagent_containers/syringe,
+		/obj/item/weapon/reagent_containers/food/snacks
+	)
+
+/obj/structure/candybucket/attackby(var/obj/item/I, var/mob/user)
+	..()
+	if(is_type_in_list(I, allowedTreats) && contents.len < candypacity)
+		if(user.drop_item(I, src))
+			to_chat(user, "<span class='notice'>You add \the [I] to \the [src].</span>")
+			updateBucket()
+	if(I.is_wrench(user))
+		I.playtoolsound(loc, 50)
+		anchored = !anchored
+
+/obj/structure/candybucket/attack_hand(var/mob/living/user)
```

Comment:
lack of sanity here, can grab candy from the bucket while you're dead or unconscious, probably through cameras if you had telekinesis

Reply:
I didn't think to test that. I assumed attack_hand had checks already. Thanks I'll add those.